src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_seckey.c

```diff
@@ -139,3 +139,82 @@ uint64_t AppleCryptoNative_SecKeyGetSimpleKeySizeInBytes(SecKeyRef publicKey)
 
     return SecKeyGetBlockSize(publicKey);
 }
+
+#if defined(TARGET_MACCATALYST) || defined(TARGET_IOS) || defined(TARGET_TVOS)
+static CFStringRef GetKeyAlgorithmIdentifier(PAL_KeyAlgorithm keyAlgorithm)
+{
+    if (keyAlgorithm == PAL_KeyAlgorithm_EC)
+        return kSecAttrKeyTypeECSECPrimeRandom;
+    if (keyAlgorithm == PAL_KeyAlgorithm_RSA)
+        return kSecAttrKeyTypeRSA;
+
+    return NULL;
+}
+
+int32_t AppleCryptoNative_SecKeyCreateWithData(uint8_t* pKey,
+                                               int32_t cbKey,
+                                               PAL_KeyAlgorithm keyAlgorithm,
+                                               int32_t isPublic,
+                                               SecKeyRef* pKeyOut,
+                                               CFErrorRef* pErrorOut)
+{
+    if (pErrorOut != NULL)
+        *pErrorOut = NULL;
+    if (pKeyOut != NULL)
+        *pKeyOut = NULL;
+
+    if (pKeyOut == NULL || pErrorOut == NULL || cbKey <= 0 || pKey == NULL)
+        return kErrorBadInput;
```

Comment:
If this is a net-new export, you can change the kErrorBadInput into asserts if you like.

```C
assert(pKeyOut != NULL);
assert(pErrorOut != NULL);
assert(cbKey > 0);
assert(pKey != NULL);

*pErrorOut = NULL;
*pKeyOut = NULL;

...
```